subsys/bluetooth/host/hci_core.c

```diff
@@ -4132,19 +4153,33 @@ static void rx_queue_put(struct net_buf *buf)
 
 static int bt_recv_unsafe(struct net_buf *buf)
 {
-	bt_monitor_send(bt_monitor_opcode(buf), buf->data, buf->len);
+	uint8_t h4_type;
+
+	if (bt_buf_get_type(buf) != BT_BUF_H4) {
```

Comment:
This seems a bit asymmetric: we don't send to the driver w/ H4 format, but we allow both `bt_buf_type` and `H4` for receiving.

Reply:
just a note, not a nak

Reply:
This is intentional. It allows us to upgrade the drivers in the RX direction without changes in the host.

Sending non-H4 to the driver in a backwards compatible way requires an API addition. I'm thinking of adding to `struct bt_hci_driver` a field `bool send_takes_h4`. But at least the name would be up for discussion, so I want to have it in a separate PR.